// ngx-responsive/esm2015/@directives/browsers/browser-info.js
import { ReplaySubject } from 'rxjs';
import { distinctUntilChanged } from 'rxjs/operators';
export class BrowserInfo {
    constructor(_responsiveState, platformService) {
        this._responsiveState = _responsiveState;
        this.platformService = platformService;
        this._isEnabledForPlatform = null;
        this.replaySubject$ = new ReplaySubject();
        this._isEnabledForPlatform = platformService.isEnabledForPlatform();
    }
    connect() {
        if (this._isEnabledForPlatform) {
            this._subscription = this._responsiveState.browser$.pipe(distinctUntilChanged())
                .subscribe((data) => {
                this._updateData(data);
            });
        }
        return this.replaySubject$.asObservable();
    }
    disconnect() {
        if (this._isEnabledForPlatform) {
            this._subscription.unsubscribe();
        }
    }
    get getBrowser() {
        return this.replaySubject$.asObservable();
    }
    _updateData(value) {
        this.replaySubject$.next(value);
    }
}
