// ngx-responsive/esm2015/@directives/browsers/browser-info.directive.js
import { __decorate } from "tslib";
/**
 * @name browser-info.directive
 * @description Browser info directive in ngx-responsive
 *
 * @license MIT
 */
import { EventEmitter, Directive, Input, Output, OnInit, OnDestroy, ChangeDetectorRef } from '@angular/core';
import { ResponsiveState } from '../../@core/providers/responsive-state/responsive-state';
import { BrowserInfo } from './browser-info';
import { PlatformService } from '../../@core/providers/platform-service/platform.service';
let BrowserInfoDirective = class BrowserInfoDirective extends BrowserInfo {
    constructor(_responsiveState, cd, platformService) {
        super(_responsiveState, platformService);
        this._responsiveState = _responsiveState;
        this.cd = cd;
        this.platformService = platformService;
        this.browser = new EventEmitter();
    }
    set browserInfo(grid_state) {
        this._updateData(this.currentstate);
    }
    ngOnInit() {
        this.connect();
    }
    ngOnDestroy() {
        this.disconnect();
    }
    _updateData(value) {
        this.browser.emit(value);
        this.cd.markForCheck();
    }
};
BrowserInfoDirective.ctorParameters = () => [
    { type: ResponsiveState },
    { type: ChangeDetectorRef },
    { type: PlatformService }
];
__decorate([
    Output()
], BrowserInfoDirective.prototype, "browser", void 0);
__decorate([
    Input()
], BrowserInfoDirective.prototype, "browserInfo", null);
BrowserInfoDirective = __decorate([
    Directive({
        selector: 'browser-info'
    })
], BrowserInfoDirective);
export { BrowserInfoDirective };
