// ngx-responsive/esm5/@directives/devices/devices-directives.js
import { __decorate, __extends } from "tslib";
/**
 * @name devices.directives
 * @description Devices directives in ngx-responsive
 *
 * @license MIT
 */
import { Directive, Input, TemplateRef, ViewContainerRef, ChangeDetectorRef } from '@angular/core';
import { ResponsiveState } from '../../@core/providers/responsive-state/responsive-state';
import { RESPONSIVE_BASE } from '../../@core/providers/responsive-base/responsive-base';
import { PlatformService } from '../../@core/providers/platform-service/platform.service';
var IsSmartTvDirective = /** @class */ (function (_super) {
    __extends(IsSmartTvDirective, _super);
    function IsSmartTvDirective(templateRef, viewContainer, _responsiveState, cd, platformService) {
        var _this = _super.call(this, templateRef, viewContainer, _responsiveState, cd, platformService) || this;
        _this._state = 'smarttv';
        _this._showWhenTrue = true;
        return _this;
    }
    Object.defineProperty(IsSmartTvDirective.prototype, "isSmartTv", {
        set: function (grid_state) {
            this.setGrid(this._state, 'device');
        },
        enumerable: true,
        configurable: true
    });
    IsSmartTvDirective.ctorParameters = function () { return [
        { type: TemplateRef },
        { type: ViewContainerRef },
        { type: ResponsiveState },
        { type: ChangeDetectorRef },
        { type: PlatformService }
    ]; };
    __decorate([
        Input()
    ], IsSmartTvDirective.prototype, "isSmartTv", null);
    IsSmartTvDirective = __decorate([
        Directive({
            selector: '[isSmartTv]'
        })
    ], IsSmartTvDirective);
    return IsSmartTvDirective;
}(RESPONSIVE_BASE));
export { IsSmartTvDirective };
var IsDesktopDirective = /** @class */ (function (_super) {
    __extends(IsDesktopDirective, _super);
    function IsDesktopDirective(templateRef, viewContainer, _responsiveState, cd, platformService) {
        var _this = _super.call(this, templateRef, viewContainer, _responsiveState, cd, platformService) || this;
        _this._state = 'desktop';
        _this._showWhenTrue = true;
        return _this;
    }
    Object.defineProperty(IsDesktopDirective.prototype, "isDesktop", {
        set: function (grid_state) {
            this.setGrid(this._state, 'device');
        },
        enumerable: true,
        configurable: true
    });
    IsDesktopDirective.ctorParameters = function () { return [
        { type: TemplateRef },
        { type: ViewContainerRef },
        { type: ResponsiveState },
        { type: ChangeDetectorRef },
        { type: PlatformService }
    ]; };
    __decorate([
        Input()
    ], IsDesktopDirective.prototype, "isDesktop", null);
    IsDesktopDirective = __decorate([
        Directive({
            selector: '[isDesktop]'
        })
    ], IsDesktopDirective);
    return IsDesktopDirective;
}(RESPONSIVE_BASE));
export { IsDesktopDirective };
var IsTabletDirective = /** @class */ (function (_super) {
    __extends(IsTabletDirective, _super);
    function IsTabletDirective(templateRef, viewContainer, _responsiveState, cd, platformService) {
        var _this = _super.call(this, templateRef, viewContainer, _responsiveState, cd, platformService) || this;
        _this._state = 'tablet';
        _this._showWhenTrue = true;
        return _this;
    }
    Object.defineProperty(IsTabletDirective.prototype, "isTablet", {
        set: function (grid_state) {
            this.setGrid(this._state, 'device');
        },
        enumerable: true,
        configurable: true
    });
    IsTabletDirective.ctorParameters = function () { return [
        { type: TemplateRef },
        { type: ViewContainerRef },
        { type: ResponsiveState },
        { type: ChangeDetectorRef },
        { type: PlatformService }
    ]; };
    __decorate([
        Input()
    ], IsTabletDirective.prototype, "isTablet", null);
    IsTabletDirective = __decorate([
        Directive({
            selector: '[isTablet]'
        })
    ], IsTabletDirective);
    return IsTabletDirective;
}(RESPONSIVE_BASE));
export { IsTabletDirective };
var IsMobileDirective = /** @class */ (function (_super) {
    __extends(IsMobileDirective, _super);
    function IsMobileDirective(templateRef, viewContainer, _responsiveState, cd, platformService) {
        var _this = _super.call(this, templateRef, viewContainer, _responsiveState, cd, platformService) || this;
        _this._state = 'mobile';
        _this._showWhenTrue = true;
        return _this;
    }
    Object.defineProperty(IsMobileDirective.prototype, "isMobile", {
        set: function (grid_state) {
            this.setGrid(this._state, 'device');
        },
        enumerable: true,
        configurable: true
    });
    IsMobileDirective.ctorParameters = function () { return [
        { type: TemplateRef },
        { type: ViewContainerRef },
        { type: ResponsiveState },
        { type: ChangeDetectorRef },
        { type: PlatformService }
    ]; };
    __decorate([
        Input()
    ], IsMobileDirective.prototype, "isMobile", null);
    IsMobileDirective = __decorate([
        Directive({
            selector: '[isMobile]'
        })
    ], IsMobileDirective);
    return IsMobileDirective;
}(RESPONSIVE_BASE));
export { IsMobileDirective };
var ShowItDeviceDirective = /** @class */ (function (_super) {
    __extends(ShowItDeviceDirective, _super);
    function ShowItDeviceDirective(templateRef, viewContainer, _responsiveState, cd, platformService) {
        var _this = _super.call(this, templateRef, viewContainer, _responsiveState, cd, platformService) || this;
        _this._showWhenTrue = true;
        return _this;
    }
    Object.defineProperty(ShowItDeviceDirective.prototype, "showItDevice", {
        set: function (grid_state) {
            this.setGrid(grid_state, 'device');
        },
        enumerable: true,
        configurable: true
    });
    ShowItDeviceDirective.ctorParameters = function () { return [
        { type: TemplateRef },
        { type: ViewContainerRef },
        { type: ResponsiveState },
        { type: ChangeDetectorRef },
        { type: PlatformService }
    ]; };
    __decorate([
        Input()
    ], ShowItDeviceDirective.prototype, "showItDevice", null);
    ShowItDeviceDirective = __decorate([
        Directive({
            selector: '[showItDevice]'
        })
    ], ShowItDeviceDirective);
    return ShowItDeviceDirective;
}(RESPONSIVE_BASE));
export { ShowItDeviceDirective };
var HideItDeviceDirective = /** @class */ (function (_super) {
    __extends(HideItDeviceDirective, _super);
    function HideItDeviceDirective(templateRef, viewContainer, _responsiveState, cd, platformService) {
        var _this = _super.call(this, templateRef, viewContainer, _responsiveState, cd, platformService) || this;
        _this._showWhenTrue = false;
        return _this;
    }
    Object.defineProperty(HideItDeviceDirective.prototype, "hideItDevice", {
        set: function (grid_state) {
            this.setGrid(grid_state, 'device');
        },
        enumerable: true,
        configurable: true
    });
    HideItDeviceDirective.ctorParameters = function () { return [
        { type: TemplateRef },
        { type: ViewContainerRef },
        { type: ResponsiveState },
        { type: ChangeDetectorRef },
        { type: PlatformService }
    ]; };
    __decorate([
        Input()
    ], HideItDeviceDirective.prototype, "hideItDevice", null);
    HideItDeviceDirective = __decorate([
        Directive({
            selector: '[hideItDevice]'
        })
    ], HideItDeviceDirective);
    return HideItDeviceDirective;
}(RESPONSIVE_BASE));
export { HideItDeviceDirective };
var IsIphoneDirective = /** @class */ (function (_super) {
    __extends(IsIphoneDirective, _super);
    function IsIphoneDirective(templateRef, viewContainer, _responsiveState, cd, platformService) {
        var _this = _super.call(this, templateRef, viewContainer, _responsiveState, cd, platformService) || this;
        _this._state = 'iphone';
        _this._showWhenTrue = true;
        return _this;
    }
    Object.defineProperty(IsIphoneDirective.prototype, "isIphone", {
        set: function (grid_state) {
            this.setGrid(this._state, 'standard');
        },
        enumerable: true,
        configurable: true
    });
    IsIphoneDirective.ctorParameters = function () { return [
        { type: TemplateRef },
        { type: ViewContainerRef },
        { type: ResponsiveState },
        { type: ChangeDetectorRef },
        { type: PlatformService }
    ]; };
    __decorate([
        Input()
    ], IsIphoneDirective.prototype, "isIphone", null);
    IsIphoneDirective = __decorate([
        Directive({
            selector: '[isIphone]'
        })
    ], IsIphoneDirective);
    return IsIphoneDirective;
}(RESPONSIVE_BASE));
export { IsIphoneDirective };
var IsIpadDirective = /** @class */ (function (_super) {
    __extends(IsIpadDirective, _super);
    function IsIpadDirective(templateRef, viewContainer, _responsiveState, cd, platformService) {
        var _this = _super.call(this, templateRef, viewContainer, _responsiveState, cd, platformService) || this;
        _this._state = 'iphone';
        _this._showWhenTrue = true;
        return _this;
    }
    Object.defineProperty(IsIpadDirective.prototype, "isIphone", {
        set: function (grid_state) {
            this.setGrid(this._state, 'standard');
        },
        enumerable: true,
        configurable: true
    });
    IsIpadDirective.ctorParameters = function () { return [
        { type: TemplateRef },
        { type: ViewContainerRef },
        { type: ResponsiveState },
        { type: ChangeDetectorRef },
        { type: PlatformService }
    ]; };
    __decorate([
        Input()
    ], IsIpadDirective.prototype, "isIphone", null);
    IsIpadDirective = __decorate([
        Directive({
            selector: '[isIpad]'
        })
    ], IsIpadDirective);
    return IsIpadDirective;
}(RESPONSIVE_BASE));
export { IsIpadDirective };
var IsAndroidMobileDirective = /** @class */ (function (_super) {
    __extends(IsAndroidMobileDirective, _super);
    function IsAndroidMobileDirective(templateRef, viewContainer, _responsiveState, cd, platformService) {
        var _this = _super.call(this, templateRef, viewContainer, _responsiveState, cd, platformService) || this;
        _this._state = 'android mobile';
        _this._showWhenTrue = true;
        return _this;
    }
    Object.defineProperty(IsAndroidMobileDirective.prototype, "isAndroidMobile", {
        set: function (grid_state) {
            this.setGrid(this._state, 'standard');
        },
        enumerable: true,
        configurable: true
    });
    IsAndroidMobileDirective.ctorParameters = function () { return [
        { type: TemplateRef },
        { type: ViewContainerRef },
        { type: ResponsiveState },
        { type: ChangeDetectorRef },
        { type: PlatformService }
    ]; };
    __decorate([
        Input()
    ], IsAndroidMobileDirective.prototype, "isAndroidMobile", null);
    IsAndroidMobileDirective = __decorate([
        Directive({
            selector: '[isAndroidMobile]'
        })
    ], IsAndroidMobileDirective);
    return IsAndroidMobileDirective;
}(RESPONSIVE_BASE));
export { IsAndroidMobileDirective };
var IsAndroidTabletDirective = /** @class */ (function (_super) {
    __extends(IsAndroidTabletDirective, _super);
    function IsAndroidTabletDirective(templateRef, viewContainer, _responsiveState, cd, platformService) {
        var _this = _super.call(this, templateRef, viewContainer, _responsiveState, cd, platformService) || this;
        _this._state = 'android tablet';
        _this._showWhenTrue = true;
        return _this;
    }
    Object.defineProperty(IsAndroidTabletDirective.prototype, "isAndroidTablet", {
        set: function (grid_state) {
            this.setGrid(this._state, 'standard');
        },
        enumerable: true,
        configurable: true
    });
    IsAndroidTabletDirective.ctorParameters = function () { return [
        { type: TemplateRef },
        { type: ViewContainerRef },
        { type: ResponsiveState },
        { type: ChangeDetectorRef },
        { type: PlatformService }
    ]; };
    __decorate([
        Input()
    ], IsAndroidTabletDirective.prototype, "isAndroidTablet", null);
    IsAndroidTabletDirective = __decorate([
        Directive({
            selector: '[isAndroidTablet]'
        })
    ], IsAndroidTabletDirective);
    return IsAndroidTabletDirective;
}(RESPONSIVE_BASE));
export { IsAndroidTabletDirective };
var IsWindowsPhoneDirective = /** @class */ (function (_super) {
    __extends(IsWindowsPhoneDirective, _super);
    function IsWindowsPhoneDirective(templateRef, viewContainer, _responsiveState, cd, platformService) {
        var _this = _super.call(this, templateRef, viewContainer, _responsiveState, cd, platformService) || this;
        _this._state = 'windows phone';
        _this._showWhenTrue = true;
        return _this;
    }
    Object.defineProperty(IsWindowsPhoneDirective.prototype, "isWindowsPhone", {
        set: function (grid_state) {
            this.setGrid(this._state, 'standard');
        },
        enumerable: true,
        configurable: true
    });
    IsWindowsPhoneDirective.ctorParameters = function () { return [
        { type: TemplateRef },
        { type: ViewContainerRef },
        { type: ResponsiveState },
        { type: ChangeDetectorRef },
        { type: PlatformService }
    ]; };
    __decorate([
        Input()
    ], IsWindowsPhoneDirective.prototype, "isWindowsPhone", null);
    IsWindowsPhoneDirective = __decorate([
        Directive({
            selector: '[isWindowsPhone]'
        })
    ], IsWindowsPhoneDirective);
    return IsWindowsPhoneDirective;
}(RESPONSIVE_BASE));
export { IsWindowsPhoneDirective };
var ShowItStandardDirective = /** @class */ (function (_super) {
    __extends(ShowItStandardDirective, _super);
    function ShowItStandardDirective(templateRef, viewContainer, _responsiveState, cd, platformService) {
        var _this = _super.call(this, templateRef, viewContainer, _responsiveState, cd, platformService) || this;
        _this._showWhenTrue = true;
        return _this;
    }
    Object.defineProperty(ShowItStandardDirective.prototype, "showItStandard", {
        set: function (grid_state) {
            this.setGrid(grid_state, 'standard');
        },
        enumerable: true,
        configurable: true
    });
    ShowItStandardDirective.ctorParameters = function () { return [
        { type: TemplateRef },
        { type: ViewContainerRef },
        { type: ResponsiveState },
        { type: ChangeDetectorRef },
        { type: PlatformService }
    ]; };
    __decorate([
        Input()
    ], ShowItStandardDirective.prototype, "showItStandard", null);
    ShowItStandardDirective = __decorate([
        Directive({
            selector: '[showItStandard]'
        })
    ], ShowItStandardDirective);
    return ShowItStandardDirective;
}(RESPONSIVE_BASE));
export { ShowItStandardDirective };
var HideItStandardDirective = /** @class */ (function (_super) {
    __extends(HideItStandardDirective, _super);
    function HideItStandardDirective(templateRef, viewContainer, _responsiveState, cd, platformService) {
        var _this = _super.call(this, templateRef, viewContainer, _responsiveState, cd, platformService) || this;
        _this._showWhenTrue = false;
        return _this;
    }
    Object.defineProperty(HideItStandardDirective.prototype, "hideItStandard", {
        set: function (grid_state) {
            this.setGrid(grid_state, 'standard');
        },
        enumerable: true,
        configurable: true
    });
    HideItStandardDirective.ctorParameters = function () { return [
        { type: TemplateRef },
        { type: ViewContainerRef },
        { type: ResponsiveState },
        { type: ChangeDetectorRef },
        { type: PlatformService }
    ]; };
    __decorate([
        Input()
    ], HideItStandardDirective.prototype, "hideItStandard", null);
    HideItStandardDirective = __decorate([
        Directive({
            selector: '[hideItStandard]'
        })
    ], HideItStandardDirective);
    return HideItStandardDirective;
}(RESPONSIVE_BASE));
export { HideItStandardDirective };
var IsPortraitDirective = /** @class */ (function (_super) {
    __extends(IsPortraitDirective, _super);
    function IsPortraitDirective(templateRef, viewContainer, _responsiveState, cd, platformService) {
        var _this = _super.call(this, templateRef, viewContainer, _responsiveState, cd, platformService) || this;
        _this._state = 'portrait';
        _this._showWhenTrue = false;
        return _this;
    }
    Object.defineProperty(IsPortraitDirective.prototype, "isPortrait", {
        set: function (grid_state) {
            this.setGrid(this._state, 'orientation');
        },
        enumerable: true,
        configurable: true
    });
    IsPortraitDirective.ctorParameters = function () { return [
        { type: TemplateRef },
        { type: ViewContainerRef },
        { type: ResponsiveState },
        { type: ChangeDetectorRef },
        { type: PlatformService }
    ]; };
    __decorate([
        Input()
    ], IsPortraitDirective.prototype, "isPortrait", null);
    IsPortraitDirective = __decorate([
        Directive({
            selector: '[isPortrait]'
        })
    ], IsPortraitDirective);
    return IsPortraitDirective;
}(RESPONSIVE_BASE));
export { IsPortraitDirective };
var IsLandscapeDirective = /** @class */ (function (_super) {
    __extends(IsLandscapeDirective, _super);
    function IsLandscapeDirective(templateRef, viewContainer, _responsiveState, cd, platformService) {
        var _this = _super.call(this, templateRef, viewContainer, _responsiveState, cd, platformService) || this;
        _this._state = 'landscape';
        _this._showWhenTrue = false;
        return _this;
    }
    Object.defineProperty(IsLandscapeDirective.prototype, "isLandscape", {
        set: function (grid_state) {
            this.setGrid(this._state, 'orientation');
        },
        enumerable: true,
        configurable: true
    });
    IsLandscapeDirective.ctorParameters = function () { return [
        { type: TemplateRef },
        { type: ViewContainerRef },
        { type: ResponsiveState },
        { type: ChangeDetectorRef },
        { type: PlatformService }
    ]; };
    __decorate([
        Input()
    ], IsLandscapeDirective.prototype, "isLandscape", null);
    IsLandscapeDirective = __decorate([
        Directive({
            selector: '[isLandscape]'
        })
    ], IsLandscapeDirective);
    return IsLandscapeDirective;
}(RESPONSIVE_BASE));
export { IsLandscapeDirective };
