// ngx-responsive/esm2015/@directives/useragent/useragent-info.directive.js
import { __decorate } from "tslib";
/**
 * @name useragent-info.directive
 * @description Useragent info directive in ngx-responsive
 *
 * @license MIT
 */
import { Output, EventEmitter, Directive, OnInit, OnDestroy, ChangeDetectorRef } from '@angular/core';
import { ResponsiveState } from '../../@core/providers/responsive-state/responsive-state';
import { UserAgentInfo } from './useragent-info';
import { PlatformService } from '../../@core/providers/platform-service/platform.service';
let UserAgentInfoDirective = class UserAgentInfoDirective extends UserAgentInfo {
    constructor(_responsiveState, cd, platformService) {
        super(_responsiveState, platformService);
        this._responsiveState = _responsiveState;
        this.cd = cd;
        this.info = new EventEmitter();
    }
    ngOnInit() {
        this.connect();
    }
    ngOnDestroy() {
        this.disconnect();
    }
    _emitUserAgent(value) {
        this.info.emit(value);
        this.cd.markForCheck();
    }
};
UserAgentInfoDirective.ctorParameters = () => [
    { type: ResponsiveState },
    { type: ChangeDetectorRef },
    { type: PlatformService }
];
__decorate([
    Output()
], UserAgentInfoDirective.prototype, "info", void 0);
UserAgentInfoDirective = __decorate([
    Directive({
        selector: 'user-agent-info'
    })
], UserAgentInfoDirective);
export { UserAgentInfoDirective };
