// ngx-responsive/esm2015/@directives/responsive-size-info/responsive-size-info.directive.js
import { __decorate } from "tslib";
/**
 * @name responsive-size-info.directive
 * @description Responsive Size Info directive in ngx-responsive
 *
 * @license MIT
 */
import { Directive, EventEmitter, Input, Output, ViewContainerRef, OnInit, OnDestroy, ChangeDetectorRef } from '@angular/core';
import { ResponsiveState } from '../../@core/providers/responsive-state/responsive-state';
import { ResponsiveSizeInfo } from './responsive-size-info';
import { PlatformService } from '../../@core/providers/platform-service/platform.service';
let ResponsiveSizeInfoDirective = class ResponsiveSizeInfoDirective extends ResponsiveSizeInfo {
    constructor(_responsiveState, viewContainer, cd, platformService) {
        super(_responsiveState, platformService);
        this._responsiveState = _responsiveState;
        this.viewContainer = viewContainer;
        this.cd = cd;
        this.statechanges = new EventEmitter();
    }
    set responsiveSizeInfo(grid_state) {
        this._updateData(this.currentstate);
    }
    ngOnInit() {
        this.connect();
    }
    ngOnDestroy() {
        this.disconnect();
    }
    _updateData(value) {
        this.statechanges.emit(value);
        this.cd.markForCheck();
    }
};
ResponsiveSizeInfoDirective.ctorParameters = () => [
    { type: ResponsiveState },
    { type: ViewContainerRef },
    { type: ChangeDetectorRef },
    { type: PlatformService }
];
__decorate([
    Input()
], ResponsiveSizeInfoDirective.prototype, "responsiveSizeInfo", null);
__decorate([
    Output()
], ResponsiveSizeInfoDirective.prototype, "statechanges", void 0);
ResponsiveSizeInfoDirective = __decorate([
    Directive({ selector: 'responsiveSizeInfo' })
], ResponsiveSizeInfoDirective);
export { ResponsiveSizeInfoDirective };
