// ngx-responsive/esm2015/@directives/devices/devices-standard-info.directive.js
import { __decorate, __param } from "tslib";
/**
 * @name devices-standard-info.directive
 * @description devices-standard-info directive in ngx-responsive
 *
 * @license MIT
 */
import { EventEmitter, Directive, Input, Output, ViewContainerRef, OnInit, OnDestroy, ChangeDetectorRef } from '@angular/core';
import { PLATFORM_ID, Inject } from '@angular/core';
import { ResponsiveState } from '../../@core/providers/responsive-state/responsive-state';
import { DeviceStandardInfo } from './devices-standard-info';
let DeviceStandardInfoDirective = class DeviceStandardInfoDirective extends DeviceStandardInfo {
    constructor(_responsiveState, viewContainer, cd, _platformId) {
        super(_responsiveState, _platformId);
        this._responsiveState = _responsiveState;
        this.viewContainer = viewContainer;
        this.cd = cd;
        this._platformId = _platformId;
        this.standard = new EventEmitter();
    }
    set deviceStandardInfo(grid_state) {
        this._updateData(this.currentstate);
    }
    ngOnInit() {
        this.connect();
    }
    ngOnDestroy() {
        this.disconnect();
    }
    _updateData(value) {
        this.standard.emit(value);
        this.cd.markForCheck();
    }
};
DeviceStandardInfoDirective.ctorParameters = () => [
    { type: ResponsiveState },
    { type: ViewContainerRef },
    { type: ChangeDetectorRef },
    { type: undefined, decorators: [{ type: Inject, args: [PLATFORM_ID,] }] }
];
__decorate([
    Input()
], DeviceStandardInfoDirective.prototype, "deviceStandardInfo", null);
__decorate([
    Output()
], DeviceStandardInfoDirective.prototype, "standard", void 0);
DeviceStandardInfoDirective = __decorate([
    Directive({ selector: 'device-standard-info' }),
    __param(3, Inject(PLATFORM_ID))
], DeviceStandardInfoDirective);
export { DeviceStandardInfoDirective };
